ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -139,6 +155,40 @@ public void run() {
   }
 
   /**
+   * Pauses hard deletes. This guarantees that any on-going hard deletes have been completed, flushed and persisted and
+   * the hard delete tokens are persisted before returning from pause. Hard delete will be in a paused state until
+   * {@link #resume()} is called.
+   */
+  void pause() throws InterruptedException, StoreException, IOException {
+    reentrantLock.lock();
+    try {
+      if (paused.compareAndSet(false, true)) {
+        logger.info("HardDelete thread has been paused ");
+        pauseCondition.signal();
```

Comment:
Wanted to avoid having many conditions. Since both these are mutually exclusive, I thought of using the same condition for both purposes. If you guys feel its confusing, I can add another condition for timedwait.